components/identity-mgt/org.wso2.carbon.identity.mgt/src/main/java/org/wso2/carbon/identity/mgt/mail/DefaultEmailSendingModule.java

```diff
@@ -73,9 +73,20 @@ public void sendEmail() {
 
         try {
             if (this.notification == null) {
-                throw new NullPointerException("Notification not set. Please set the notification before sending messages");
+                throw new IllegalStateException("Notification not set. " +
+                        "Please set the notification before sending messages");
             }
             PrivilegedCarbonContext.startTenantFlow();
+            if (notificationData != null) {
+                String tenantDomain = notificationData.getDomainName();
+                PrivilegedCarbonContext carbonContext = PrivilegedCarbonContext.getThreadLocalCarbonContext();
+                carbonContext.setTenantDomain(tenantDomain);
+                carbonContext.getTenantId(true);
```

Comment:
Changed to carbonContext.setTenantDomain(tenantDomain, true) in carbonContext.setTenantDomain(tenantDomain, true)